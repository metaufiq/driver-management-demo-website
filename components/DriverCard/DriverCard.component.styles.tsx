import styled from "styled-components";

export const Container = styled.div`
  padding: 1rem;
  background-color: white;
  border-radius: 0.125rem;
`

export const Title = styled.p`
  color: #c6c6c6;
  font-size: 0.75rem;
  margin: 0;
`

export const Description = styled.p`
  margin: 0;
`

export const ContentContainer = styled.div`
  padding: 0.5rem 0;
`