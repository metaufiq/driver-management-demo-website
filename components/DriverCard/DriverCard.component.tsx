import { Component } from "../../index.types";
import { Avatar, Container, ContentContainer, Description, Title } from "./DriverCard.component.styles";

import { Props } from "./DriverCard.component.types";

const DriverCard:Component<Props> = ({user})=>{
  return(
  <Container>
    <ContentContainer>
      <Avatar
        src={user.profilePicture}
      />
    </ContentContainer>
    <ContentContainer>
      <Title>Nama Driver</Title>
      <Description>{user.firstName}, {user.lastName}</Description>
    </ContentContainer>
    <ContentContainer>
      <Title>Telepon</Title>
      <Description>{user.phone}</Description>
    </ContentContainer>
    <ContentContainer>
      <Title>Email</Title>
      <Description>{user.email}</Description>
    </ContentContainer>
    <ContentContainer>
      <Title>Tanggal Lahir</Title>
      <Description>{user.dob}</Description>
    </ContentContainer>
  </Container>)
}

export default DriverCard;