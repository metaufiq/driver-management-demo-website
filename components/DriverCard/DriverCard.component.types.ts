import { User } from "../../index.types"

export type Props = {
  user: User
}