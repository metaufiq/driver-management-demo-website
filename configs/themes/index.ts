const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };