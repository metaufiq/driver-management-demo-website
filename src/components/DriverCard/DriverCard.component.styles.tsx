import styled from "styled-components";

export const Container = styled.div`
  background-color: white;
  border-radius: 0.125rem;
  display:inline-block;
`

const GrayText = styled.p`
  color: #c6c6c6;
`

export const Header = styled.div`
  padding: 1rem;
  border-bottom: 0.5px solid gray;
  display: flex;
  flex-direction: row;
  gap: 1rem;
  justify-content: space-between;
  align-items: center;
`

export const DriverIDTitle = styled(GrayText)`
  display: inline-block;
  margin-right: 0.5rem;
  margin-top: 0;
  margin-bottom: 0;
  font-size: 0.8rem;
`
export const DriverIDValue = styled.p`
  display: inline-block;
  margin-top: 0;
  color: ${({theme})=>theme.colors.primary};
  margin-bottom: 0;
  font-size: 0.8rem;
`

export const ThreeDotsContainer = styled.div`
  color: #c6c6c6;
  align-items: center;
  justify-content: center;
  display: flex;
`

export const Title = styled(GrayText)`
  font-size: 0.5rem;
  margin: 0;
`

export const Description = styled.p`
  margin: 0;
  font-size: 0.8rem;
`

export const ContentContainer = styled.div`
  padding: 0.5rem 1rem;
  display: flex;
  flex-direction: column;
`

