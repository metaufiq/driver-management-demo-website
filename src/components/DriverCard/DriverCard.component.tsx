import { HiDotsHorizontal } from "react-icons/hi";

import { Component } from "../../../index.types";
import Avatar from "../Avatar";
import { 
  Container, 
  ContentContainer,
  Description,
  DriverIDTitle,
  DriverIDValue, 
  Header,
  ThreeDotsContainer, 
  Title 
} from "./DriverCard.component.styles";

import { Props } from "./DriverCard.component.types";

const DriverCard:Component<Props> = ({user})=>{
  return(
  <Container>
    <Header>
      <div>
        <DriverIDTitle>Driver ID</DriverIDTitle>
        <DriverIDValue>{user.username}</DriverIDValue>
      </div>
      <ThreeDotsContainer>
        <HiDotsHorizontal/>
      </ThreeDotsContainer>
    </Header>
    <Avatar
      src={user.profilePicture}
    />
    <ContentContainer>
      <Title>Nama Driver</Title>
      <Description>{user.firstName}, {user.lastName}</Description>
    </ContentContainer>
    <ContentContainer>
      <Title>Telepon</Title>
      <Description>{user.phone}</Description>
    </ContentContainer>
    <ContentContainer>
      <Title>Email</Title>
      <Description>{user.email}</Description>
    </ContentContainer>
    <ContentContainer>
      <Title>Tanggal Lahir</Title>
      <Description>{user.dob}</Description>
    </ContentContainer>
  </Container>)
}

export default DriverCard;