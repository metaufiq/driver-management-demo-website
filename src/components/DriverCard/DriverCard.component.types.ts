import { User } from "@dmd/types"

export type AvatarProps ={
  picture: User['picture']
}

export type Props = {
  user: User
}