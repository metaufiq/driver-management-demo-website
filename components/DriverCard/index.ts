import DriverCard from "./DriverCard.component";

export default DriverCard