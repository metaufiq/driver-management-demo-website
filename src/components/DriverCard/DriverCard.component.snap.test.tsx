import { assertSnapshots } from "@dmd/test";
import DriverCard from ".";
import { Configs } from "@dmd/test/index.types";
import { Props } from "./DriverCard.component.types";
import { themes } from "../../configs/themes";

describe('DriverCardSnapshot', () => {

  const configs: Configs<Props> = [
    {
      desc: 'should render normal driver card',
      props:{
        user:{
          dob: '12/10/1998',
          email: 'email@email.com',
          firstName: 'firstName',
          lastName: 'lastName',
          phone: '098123123123',
          username: 'userName',
          picture: {
            large: 'large_picture_url',
            medium: 'medium_picture_url',
            thumbnail: 'thumbnail_picture_url',
          }
        }
      },
      theme: themes,
      useTheme: true
    }
  ];

  assertSnapshots(DriverCard, configs);
});
