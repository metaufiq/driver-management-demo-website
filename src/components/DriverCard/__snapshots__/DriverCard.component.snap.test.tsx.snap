// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriverCardSnapshot should render normal driver card 1`] = `
<div
  className="sc-bczRLJ bZLAfS"
>
  <div
    className="sc-dkzDqf kwBbVE"
  >
    <div>
      <p
        className="sc-gsnTZi sc-hKMtZM FhkDi eciUyH"
      >
        Driver ID
      </p>
      <p
        className="sc-eCYdqJ eLaVsl"
      >
        userName
      </p>
    </div>
    <div
      className="sc-jSMfEi bKWWQD"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-ftvSup cqdCPy"
  >
    <div
      className="sc-iqcoie dHgEIj"
    />
    <div
      className="sc-papXJ jhdeYe"
    >
      <div
        className="sc-jqUVSM ifauud"
      >
        <p
          className="sc-gsnTZi sc-gKXOVf FhkDi beJtKk"
        >
          Nama Driver
        </p>
        <p
          className="sc-iBkjds bthwDz"
        >
          firstName
          , 
          lastName
        </p>
      </div>
      <div
        className="sc-jqUVSM ifauud"
      >
        <p
          className="sc-gsnTZi sc-gKXOVf FhkDi beJtKk"
        >
          Telepon
        </p>
        <p
          className="sc-iBkjds bthwDz"
        >
          098123123123
        </p>
      </div>
      <div
        className="sc-kDDrLX bbZLZe"
      >
        <div
          className="sc-jqUVSM ifauud"
        >
          <p
            className="sc-gsnTZi sc-gKXOVf FhkDi beJtKk"
          >
            Email
          </p>
          <p
            className="sc-iBkjds bthwDz"
          >
            email@email.com
          </p>
        </div>
        <div
          className="sc-jqUVSM ifauud"
        >
          <p
            className="sc-gsnTZi sc-gKXOVf FhkDi beJtKk"
          >
            Tanggal Lahir
          </p>
          <p
            className="sc-iBkjds bthwDz"
          >
            12/10/1998
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
